### PR TITLE
Panic fixes

### PR DESCRIPTION
**Pull request by Zamony**
- [ ] Link to issue, e.g. `Resolves #NNN`
 - [ ] Documentation added (if applicable)
 - [ ] Tests added
 - [ ] Branch rebased on top of current main (`git pull --rebase origin main`)
 - [ ] Changes squashed to a single commit (described [here](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
 - [x] Build is green in Travis CI
 - [x] You have certified that the contribution is your original work and that you license the work to the project under the [Apache 2 license](https://github.com/nats-io/nats-server/blob/main/LICENSE)

Resolves panics in the code.

### Changes proposed in this pull request:

 - This PR fixes some of the panics in the code

### CODE REVIEW

File: server/route.go [hunk 1]
```diff
@@ -1949,12 +1950,12 @@ func (c *client) processRouteConnect(srv *Server, arg []byte, lang string) error
 		c.closeConnection(WrongGateway)
 		return ErrWrongGateway
 	}
-	var perms *RoutePermissions
-	//TODO this check indicates srv may be nil. see srv usage below
-	if srv != nil {
-		perms = srv.getOpts().Cluster.Permissions
+
+	if srv == nil {
+		return errors.New("server is not set")
```
Comment (derekcollison) [inline server/route.go:1955]: Maybe use `ErrServerNotRunning` or add a new item in errors.go.

File: server/stream.go [hunk 2]
```diff
@@ -1788,8 +1788,12 @@ func gatherSourceMirrorSubjects(subjects []string, cfg *StreamConfig, acc *Accou
 
 // Return the subjects for a stream source.
 func (a *Account) streamSourceSubjects(ss *StreamSource, seen map[string]bool) (subjects []string, hasExt bool) {
-	if ss != nil && ss.External != nil {
-		return nil, true
+	if ss == nil {
+		return nil, false
+	} else {
```
Comment (derekcollison) [inline server/stream.go:1793]: Make on line?

`else if ss.External != nil {`

Comment (Zamony) [reply server/stream.go]: Thanks. Changed to this.

File: server/route.go (hunk 1 above)
Comment (Zamony) [reply server/route.go]: Thanks. Changed to `ErrServerNotRunning`

**Review by derekcollison — APPROVED** [review]
LGTM!

**Pull request merged**